packages/shared/src/hooks/useReadingRank.ts

```diff
@@ -115,19 +113,22 @@ export default function useReadingRank(): ReturnType {
     }
   };
 
+  // Let the rank update and then show progress animation, slight delay so the user won't miss it
+  const displayProgress = () => setTimeout(updateShownProgress, 300);
+
   useEffect(() => {
     if (remoteRank && loadedCache) {
       if (
         !cachedRank ||
-        remoteRank.rank.progressThisWeek < cachedRank.rank.progressThisWeek ||
-        !cachedRank.rank.rankLastWeek
+        remoteRank.rank.progressThisWeek < cachedRank.rank.progressThisWeek
       ) {
         cacheRank();
-        return;
+      } else if (cachedRank && !cachedRank.rank.rankLastWeek) {
+        cacheRank();
+        displayProgress();
```

Comment:
Why do we need this case?
I'd suggest adding a comment above every case in this use effect. It's somewhat confusing

Reply:
After checking [this PR](https://github.com/dailydotdev/apps/pull/736), the condition `!cachedRank.rank.rankLastWeek` was added to fix a weird animation behavior. To support the case I kept its sole intention of updating the cache. But then the block statement where it was included, it returns immediately, it results in the progress not being shown for users on the first week that should be valid for animation - like they've read 4 consecutive days but since the `rank.rankLastWeek` was `null` they skipped the animation.

Reply:
After checking this PR, the condition !cachedRank.rank.rankLastWeek was added to fix a weird animation behavior. To support the case I kept its sole intention of updating the cache. But then the block statement where it was included, it returns immediately, it results in the progress not being shown for users on the first week that should be valid for animation - like they've read 4 consecutive days but since the rank.rankLastWeek was null they skipped the animation.

Reply:
After checking [this PR](https://github.com/dailydotdev/apps/pull/736), the condition `!cachedRank.rank.rankLastWeek` was added to fix a weird animation behavior. To support the case I kept its sole intention of updating the cache. But then the block statement where it was included, it returns immediately, it results in the progress not being shown for users on the first week that should be valid for animation - like they've read 4 consecutive days but since the rank.rankLastWeek was `null` they skipped the animation.

Reply:
Let me add the comments now.